src/h_vialib/secure/secrets.py

```diff
@@ -0,0 +1,49 @@
+import base64
+import json
+from typing import Tuple
+
+from Cryptodome import Random
+from Cryptodome.Cipher import AES
+from Cryptodome.Util.Padding import pad
+
+
+class SecureSecrets:
+    def __init__(self, secret: bytes):
+        self._secret = pad(secret, 16)
+
+    def encrypt_dict(self, payload: dict) -> str:
+        """
+        Encrypt a dictionary using AES.
+
+        The returned string is a json encoded dictionary containing both the AES IV
+        and the cipher text.
+        """
+        dict_json = json.dumps(payload).encode("utf-8")
+        aes_iv, encrypted_json = self._encrypt(dict_json)
+
+        return json.dumps(
+            {
+                "iv": base64.urlsafe_b64encode(aes_iv).decode("utf-8"),
+                "payload": base64.urlsafe_b64encode(encrypted_json).decode("utf-8"),
+            }
+        )
+
+    def decrypt_dict(self, payload: str) -> dict:
+        """Decypts payloads created by `encrypt_dict`."""
+        payload_dict = json.loads(payload)
+
+        aes_iv = payload_dict.get("iv", "")
+        cipher = payload_dict.get("payload", "")
+
+        aes_iv = base64.urlsafe_b64decode(aes_iv)
+        cipher = base64.urlsafe_b64decode(cipher)
+
+        return json.loads(self._decrypt(aes_iv, cipher))
+
+    def _decrypt(self, aes_iv, encrypted) -> bytes:
```

Comment:
Making these private. We might need them in the future but as this is a library we want to keep the interface under control.

Reply:
So the existing `SecureToken` and `SecureURL` classes aren't appropriate for this use-case? I guess not because it looks like `SecureURL` only signs the URL doesn't encrypt it.

I wonder if we should rename some of these classes, to clarify what they do and don't do? For example perhaps `SecureURL` should be `SignedURL` if it just signs and doesn't encrypt. I think `SecureSecrets` encrypts a dict but doesn't sign it, so perhaps `EncryptedDict`?

I think we should probably stop using the words "secret" and "secure" because it's not clear exactly what they mean, and maybe instead say "signed" and "encrypted".

Reply:
EncryptedDict is a better name :+1:.

Reply:
I wrote this as `EncryptedDict` but from the callers perspetive it looked like it was as `dict` subclass supporting encryption. It didn't feel right.

I renamed the class to a more explicit `Encryption` and keep the method names.